share/nisar/examples/noise_spectra_analysis.py

```diff
@@ -0,0 +1,501 @@
+"""
+Analyze noise-only receive power spectra. Collect spectral information with regards to
+RFI in terms of power and frequency locations.
+"""
+
+import numpy as np
+import argparse
+import h5py
+from scipy.fft import fft, fftshift
+from scipy.signal import welch
+from numpy import linalg as la
+from nisar.products.readers.Raw import Raw
+from collections.abc import Iterator
+import warnings
+import os
+
+
+def cmd_line_parse():
+    parser = argparse.ArgumentParser(description='Perform spectra analysis on input L0B file.')
+
+    parser.add_argument(
+        '-i',
+        '--input', 
+        dest='input_file', 
+        type=str,
+        required=True, 
+        help='Input L0B file'
+    )
+
+    parser.add_argument(
+        '-o',
+        '--output', 
+        dest='output_dir', 
+        type=str,
+        required=True, 
+        help='Output data path. Output file name is input file name + SPECTRA.h5'
+    )
+
+    parser.add_argument(
+        "-c",
+        "--cpi",
+        dest="cpi_len",
+        type=int,
+        required=False,
+        default="6",
+        help="Number of pulses in a coherent processing interval. Default: 6",
+    )
+
+    parser.add_argument(
+        "-n",
+        "--nfft",
+        dest="num_fft",
+        type=int,
+        default="2048",
+        help="Number of samples per segment for Scipy Welch function." 
+             "It is equal to number of FFT to be performed on each segment."
+             "Default: 2048"
+    )
+
+    parser.add_argument(
+        "-b",
+        "--blk_size",
```

Comment:
```suggestion
        "--block-size",
```
I prefer we avoid mixing dashes and underscores in CLI options. Also we may as well spell out "block" since lazy typists can just use `-b` instead.